### PR TITLE
fix(iOS): SimpleOrientationSensor activation

### PR DESCRIPTION
**Pull request by ajpinedam**
GitHub Issue (If applicable): closes #

## PR Type

What kind of change does this PR introduce?
- Bugfix

### <samp>🤖 Generated by Copilot at c1698b5</samp>

Improved `SimpleOrientationSensor` support on iOS by enabling nullable reference types, handling device motion better, and adding logging and error handling.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UWP/Devices/Sensors/SimpleOrientationSensor.iOS.cs [hunk 1]
```diff
@@ -8,20 +9,40 @@ namespace Windows.Devices.Sensors
...
-				this.Log().Error("DeviceMotion is available");
+				this.Log().Info("DeviceMotion is available");
 				_motionManager.DeviceMotionUpdateInterval = _updateInterval;
+
 				_motionManager.StartDeviceMotionUpdates(operationQueue, (motion, error) =>
 				{
+					// Motion and Error can be null: https://developer.apple.com/documentation/coremotion/cmdevicemotionhandler
+					if (error is not null)
+					{
+						this.Log().Error($"SimpleOrientationSensor returned error when reading Device Motion updates. {error.Description}");
```
Comment (jeromelaban) [inline src/Uno.UWP/Devices/Sensors/SimpleOrientationSensor.iOS.cs:36]: Use `this.Log().IsEnabled` before logging

File: src/Uno.UWP/Devices/Sensors/SimpleOrientationSensor.iOS.cs [hunk 2]
```diff
@@ -8,20 +9,40 @@ namespace Windows.Devices.Sensors
...
+				// We should not start more than one CMMotionManager to avoid performance hit.
+				_motionManager.StopDeviceMotionUpdates();
+			}
+
 			_motionManager = new CMMotionManager();
 			if (_motionManager.DeviceMotionAvailable) // DeviceMotion is not available on all devices. iOS4+
 			{
 				var operationQueue = (NSOperationQueue.CurrentQueue == null || NSOperationQueue.CurrentQueue == NSOperationQueue.MainQueue) ? new NSOperationQueue() : NSOperationQueue.CurrentQueue;
-				this.Log().Error("DeviceMotion is available");
+				this.Log().Info("DeviceMotion is available");
```
Comment (jeromelaban) [inline src/Uno.UWP/Devices/Sensors/SimpleOrientationSensor.iOS.cs:28]: Same here

File: src/Uno.UWP/Devices/Sensors/SimpleOrientationSensor.iOS.cs [hunk 3]
```diff
@@ -8,20 +9,40 @@ namespace Windows.Devices.Sensors
...
+		private static CMMotionManager? _motionManager;
 		private const double _updateInterval = 0.5;
 		private const double _threshold = 0.5;
 
 		partial void Initialize()
 		{
+			if (_motionManager is { DeviceMotionAvailable: true })
+			{
+				// We should not start more than one CMMotionManager to avoid performance hit.
+				_motionManager.StopDeviceMotionUpdates();
```
Comment (jeromelaban) [inline src/Uno.UWP/Devices/Sensors/SimpleOrientationSensor.iOS.cs:21]: In theory, we should not have multiple instances of this class (it's a singleton) but I wonder, how did you fall in this case?

**Review by jeromelaban — APPROVED** [review]

File: doc/articles/features/orientation-sensor.md [hunk 4]
```diff
@@ -5,34 +5,34 @@ uid: Uno.Features.OrientationSensor
...
 | `GetDefault`         | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ |
-| `ReadingChanged` | ✔ | ✔ | ✔ | ✔ | ✖ | ✖| ✖ |
+| `OrientationChanged` | ✔ | ✔ | ✔ | ✔ | ✖ | ✖| ✖ |
 
 ## Using SimpleOrientationSensor with Uno
- 
- * The `GetDefault` method is available on all targets and will return `null` on those which do not support `SimpleOrientationSensor` or devices that do not have such a sensor.
- * Ensure to unsubscribe from the `ReadingChanged` event when you no longer need the readings, so that the sensor is no longer active to avoid unnecessary battery consumption.
+
+* The `GetDefault` method is available on all targets and will return `null` on those that do not support `SimpleOrientationSensor` or devices that do not have such a sensor.
```
Comment (jeromelaban) [inline doc/articles/features/orientation-sensor.md:23]: Maybe we can explicitly say that the simulator doe not support it?

```suggestion
* The `GetDefault` method is available on all targets and will return `null` on those that do not support `SimpleOrientationSensor` or devices that do not have such a sensor (e.g. the iOS Simulator).
```

Comment (ajpinedam) [reply doc/articles/features/orientation-sensor.md:23]: The thing is that `GetDefault()` still returns an instance, not null,  but the `OnOrientationChanged` event never triggers. Making `GetDefault()` return null on devices that do not support (including the simulators) is the right move, but it's a breaking change. 

@jeromelaban what's your suggestion?

Comment (jeromelaban) [reply doc/articles/features/orientation-sensor.md:23]: That's indeed interesting. We can make that change for Uno 5, we already have other related breaking changes.

Comment (ajpinedam) [reply doc/articles/features/orientation-sensor.md:23]: Copied!

**Pull request merged**